src/history/hs.c

```diff
@@ -564,12 +569,17 @@ __wt_hs_insert_updates(WT_CURSOR *cursor, WT_BTREE *btree, WT_RECONCILE *r, WT_M
             if (prev_upd->type == WT_UPDATE_TOMBSTONE) {
                 WT_ASSERT(session, modifies.size > 0);
                 __wt_modify_vector_pop(&modifies, &prev_upd);
-
-                /* The update newer to a TOMBSTONE must be a full update. */
-                WT_ASSERT(session, prev_upd->type == WT_UPDATE_STANDARD);
-            }
-
-            if (prev_upd->type == WT_UPDATE_MODIFY) {
+                /* The base value of a modfiy newer than a tombstone is the empty value. */
+                if (prev_upd->type == WT_UPDATE_MODIFY) {
```

Comment:
Is it at all possible to break out this logic into a function or macro as we have it in two separate places? Argueably three as we do another if upd->type == WT_UPDATE_MODIFY below this too. Also if its not please remove the space on line 578.